### DISCUSSION TITLE
Plot multiple spheres in a heatmap using pyvista

### DISCUSSION THREAD

**Discussion by pranoy-ray** [pyvista/pyvista#6226] [Q&A]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
I have a 3D numpy array (in the form of a cube). I have multiple zeros in it which I want to make transparent. The others I would like to show using the heatmap.
```

### Description

To provide greater visibility, I am sharing the code and a diagram. The spheres refer to multiple atoms which I would like to use as spheres.

### System Information

```shell
#Dual PyVista 3D numpy array plot code
def dualplot(data1,data2,file1="     ",file2="     ",title="     "):
    #setting theme
    pv.set_plot_theme("document")
    #setting title
    pv.global_theme.title = title

    #Plot 1: Sliced Heatmap
    # Create the spatial reference
    grid1 = pv.UniformGrid()
    # Set the grid dimensions: shape + 1 because we want to inject our values on the CELL data
    grid1.dimensions = np.array(np.shape(data1)) + 1
    # Edit the spatial reference
    grid1.spacing = (1, 1, 1)  # These are the cell sizes along each axis
    # Add the data values to the cell data
    grid1.cell_data[file1] = data1.flatten(order="F")  # Flatten the array!

    grid1=grid1.slice_orthogonal()
    #slices.plot()
    # Now plot the grid!
    #grid1.plot(show_edges=False)

    #Plot 2: Boxed Heatmap
    grid2 = pv.UniformGrid()
    # Set the grid dimensions: shape + 1 because we want to inject our values on the CELL data
    grid2.dimensions = np.array(np.shape(data2)) + 1
    # Edit the spatial reference
    grid2.spacing = (1, 1, 1)  # These are the cell sizes along each axis
    # Add the data values to the cell data
    grid2.cell_data[file2] = data2.flatten(order="F")  # Flatten the array!
    # Now plot the grid!
    #grid2.plot(show_edges=False)

    #Dual Plot
    # Create a common colorbar for both subplots
    cmap = "coolwarm"
    clim = [0, 1]
    common_bar_args = dict(
        n_labels=5,
        position_x=1.9,
        position_y=0.15,
        width=0.02,
        height=0.7,
        label_font_size=16,
        font_family="arial",
        color="black",
    )

    common_bar_args = dict(
        n_labels=5,
        width=0.02,
        height=0.7,
        label_font_size=16,
        font_family="arial",
        color="black",
    )

    from matplotlib.colors import ListedColormap
    cmap = plt.cm.RdBu
    my_cmap = cmap(np.arange(cmap.N))
    my_cmap[:, -1] = np.linspace(0, 1, cmap.N)
    my_cmap = ListedColormap(my_cmap)

    max_val = 4
    transparency_ticks = 50
    colors = [mpl.colors.hsv_to_rgb((0.83, a/transparency_ticks, 1)) 
            for a in range(transparency_ticks)]
    cmap = mpl.colors.ListedColormap(colors)
    norm = mpl.colors.Normalize(vmin=0, vmax=max_val)

    # Create a Plotter object and add the subplots
    plotter = pv.Plotter(shape=(1, 2), border=False)
    plotter.subplot(0, 0)
    plotter.add_mesh(grid1, cmap=plt.cm.RdBu, opacity="linear")#.warp_by_scalar())#, cmap=cmap, clim=clim)
    plotter.subplot(0, 1)
    plotter.add_mesh(grid2, style="points", smooth_shading=True)#.warp_by_scalar())#, cmap=cmap, clim=clim)

    # Add the common colorbar
    #plotter.add_scalar_bar(**common_bar_args)

    plotter.show()
```

### Screenshots

<img width="1187" alt="Screenshot 2024-06-07 at 5 52 05 PM" src="https://github.com/pyvista/pyvista/assets/80657118/8cda8398-32a0-4114-9bfc-0bc99eb6efa5">

**Comment by MatthewFlamm** [comment]
There are probably 10-20 different flavors to accomplish this.  It comes down to visualization preference and meaning of the dataset.  https://docs.pyvista.org/version/stable/examples/99-advanced/atomic-orbitals.html has a few examples you could start from.

https://docs.pyvista.org/version/dev/examples/02-plot/ghost-cells.html might be useful to look at to remove `ImageData` blank data.  Note that here I've linked to a dev version of the documentation because the stable version example is wrong.  I believe the code is fine in the stable version, but it is the example in the docs which is incorrect.
